.github/workflows/build-os-image.yml

```
@@ -79,41 +93,65 @@ jobs:
         id: version
         uses: ./.github/actions/pseudo_version
 
+      - name: Determine ref
+        id: ref
+        run: |
+          if [ "${{ inputs.isRelease }}" = "true" ]; then
```

```suggestion
          if [[ "${{ inputs.isRelease }}" = "true" ]]; then
```
In general, try using `[[` instead of `[`